src/calliope/io.py

```diff
@@ -39,26 +40,33 @@ def read_netcdf(path):
 
 
 def _pop_serialised_list(
-    attrs: dict, serialised_items: Union[str, list, np.ndarray]
+    attrs: dict, serialised_items: Union[str, list]
 ) -> Union[list, np.ndarray]:
+    """Pop a list of serialised attributes from the attribute dictionary."""
     serialised_ = attrs.pop(serialised_items, [])
-    if not isinstance(serialised_, (list, np.ndarray)):
-        return [serialised_]
-    else:
-        return serialised_
+    return listify(serialised_)
 
 
 def _serialise(attrs: dict) -> None:
     """Convert troublesome datatypes to nicer ones in xarray attribute dictionaries.
 
     This will tackle dictionaries (to string), booleans (to int), None (to string), and sets (to list).
 
+    We also make note of any single element lists as they will be stored as simple strings in netcdf format,
+    but we want them to be returned as lists on loading the data from file.
+
     Args:
         attrs (dict):
             Attribute dictionary from an xarray Dataset/DataArray.
             Changes will be made in-place, so be sure to supply a copy of your dictionary if you want access to its original state.
     """
-    # Convert dicts attrs to yaml strings
+    # Keep track of single element lists, to listify them again when loading from file.
```

Comment:
Shouldn't we instead make sure to serialize all lists regardless of their length?
NetCDF does not support lists in attributes anyways. The single case was just what triggered this error.

Reply:
Maybe just highlighting that a list was serialized should be enough?
This code would work, but I wonder if just noting serialization for all lists would be a more generic, leaner fix.

Reply:
Hmm, so xarray is already doing some magic to serialise lists before saving to netcdf? Because we didn't have problems with saving lists in the past, only sets/tuples. Lists is something it can seemingly handle

Reply:
So, xarray has handled lists of strings via its own encoding method as of [v0.10.4](https://docs.xarray.dev/en/stable/whats-new.html#v0-10-4-16-may-2018). So I'd continue with the current approach of doing the minimum necessary to make xarray happy.

Reply:
Understood. Changes approved!